### PR TITLE
AzureRT S53 - Set/Get Diagnostics Profile & Log

### PR DESCRIPTION
**Pull request by huangpf**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@huangpf__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (phuang). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: lib/commands/arm/vm/vm._js [hunk 1]
```diff
@@ -98,6 +98,10 @@ exports.init = function (cli) {
     .option('-s, --subscription <subscription>', $('the subscription identifier'))
     .option('--tags <tags>', $('Tags to set to the virtual machine. Can be multiple. ' +
             'In the format of \'name=value\'. Name is required and value is optional. For example, --tags tag1=value1;tag2'))
+    .option('--enable-boot-diagnostics <enable-boot-diagnostics>', $('To enable or disable the boot diagnostics for VM. ' +
```
Comment (amarzavery) [inline lib/commands/arm/vm/vm._js]: Why do we want the user to provide a value like true or false?
How about the following experience:
- By default the boot-diagnostics is disabled. If the user wants it then he should just specify the switch --enable-boot-diagnostics.

Look at the following example in rediscache create command. 
https://github.com/Azure/azure-xplat-cli/blob/dev/lib/commands/arm/rediscache/rediscache._js#L60

File: lib/commands/arm/vm/vmClient._js [hunk 2]
```diff
@@ -899,6 +994,44 @@ __.extend(VMClient.prototype, {
...
+        accountName : null,
+        containerName : null,
+        blobName : null
+    };
+
+    var parsedResult = url.parse(serialConsoleLogBlobUri, true);
+    var hostName = parsedResult.host || '';
+    var hostSegments = hostName.split('.');
+    var containerAndBlobPath = parsedResult.path || '';
+    if (containerAndBlobPath.length > 0 && containerAndBlobPath.charAt(0) == '/') {
```
Comment (amarzavery) [inline lib/commands/arm/vm/vmClient._js]: use === for comparison in javascript
http://stackoverflow.com/questions/359494/does-it-matter-which-equals-operator-vs-i-use-in-JavaScript-comparisons

File: lib/commands/arm/vm/vmClient._js [hunk 3]
```diff
@@ -362,6 +377,86 @@ __.extend(VMClient.prototype, {
...
+        if (instanceView && instanceView.bootDiagnostics && instanceView.bootDiagnostics.serialConsoleLogBlobUri)
+        {
+          var serialConsoleLogBlobUri = instanceView.bootDiagnostics.serialConsoleLogBlobUri;
+          this.cli.output.info(util.format($('Serial Console Log Blob Uri:\n%s'), serialConsoleLogBlobUri));
+
+          //Init StorageUtil
+          StorageUtil.init(this.cli);
+
+          var result = this._getStorageAccountContainerAndBlobFromUri(serialConsoleLogBlobUri);
+          this.cli.output.info(util.format($('serialConsoleLogBlobUri = "%s"'), serialConsoleLogBlobUri));
```
Comment (amarzavery) [inline lib/commands/arm/vm/vmClient._js]: Why would you want to the output the same info in two different formats:
- **Serial Console Log Blob Uri**
-  **serialConsoleLogBlobUri**

**Comment by amarzavery** [comment]
@huangpf - Everything else looks good. When do you plan to move the compute node sdk from lib/commands/arm/armsdk to the actual package "azure-arm-compute" package that resides in node sdk?

File: lib/commands/arm/vm/vmClient._js (hunk 3 above)
Comment (huangpf) [inline lib/commands/arm/vm/vmClient._js]: Duplicate code - I would remove the second one...

**Comment by huangpf** [comment]
Thanks @amarzavery. I would address your comments and create an item to follow up your request of having the actual package of compute by the next PR.

File: lib/commands/arm/armsdk/compute.js [hunk 4]
```diff
@@ -20,16 +20,16 @@ var ComputeManagementClient = require('./computeManagementClient');
 exports.ComputeManagementClient = ComputeManagementClient;
```
Comment (yugangw-msft) [inline lib/commands/arm/armsdk/compute.js]: please remove this file and rather reference the latest compute node sdk lib

Comment (yugangw-msft) [inline lib/commands/arm/armsdk/compute.js]: and also remove the "/computeManagementClient.js" below

File: lib/commands/arm/vm/vmClient._js [hunk 5]
```diff
@@ -30,8 +30,20 @@ var VirtualMachine = require('./virtualMachine');
 var AvailabilitySet = require('./../availabilityset/availabilitySet');
 var VMImage = require('./vmImage');
 
+var StorageUtil = require('../../../util/storage.util');
+var url = require('url');
```
Comment (yugangw-msft) [inline lib/commands/arm/vm/vmClient._js]: please move the url reference to top. The sequence of 'require' is
1. Node build-in lib
2. 3rd path lib
3. Ours
For each category, sort by alphabetic

File: lib/commands/arm/vm/vm._js [hunk 6]
```diff
@@ -318,10 +336,13 @@ exports.init = function (cli) {
             'In the format of \'name=value\'. Name is required and value is optional. For example, -t tag1=value1;tag2'))
     .option('-T, --no-tags', $('Remove all tags'))
     .option('-s, --subscription <subscription>', $('the subscription identifier'))
+    .option('--disable-boot-diagnostics', $('The switch parameter to disable the boot diagnostics for VM.'))
+    .option('--enable-boot-diagnostics', $('The switch parameter to enable the boot diagnostics for VM.'))
```
Comment (amarzavery) [inline lib/commands/arm/vm/vm._js]: The vm create command uses one switch --enable-boot-diagnostics whereas the vm set command uses two switches
--enable-boot-diagnostics and --disable-boot-diagnostics

- Why do you want to provide two separate switches for a task that can be accomplished in one switch?

**Pull request merged**